aea/protocols/dialogue/base.py

```diff
@@ -857,66 +851,119 @@ def _basic_validation_non_initial_message(
                 ),
             )
 
-        last_message_id = self.last_message.message_id  # type: ignore
-        if message_id != last_message_id + 1:
-            return (
-                False,
-                "Invalid message_id. Expected {}. Found {}.".format(
-                    last_message_id + 1, message_id
-                ),
-            )
-        if target < 1:
-            return (
-                False,
-                "Invalid target. Expected a value greater than or equal to 1. Found {}.".format(
-                    target
-                ),
+        # it's non initial check, assume there some messages in the dialogue
+        if self.is_empty:
+            return False, "Can not reply cause no messages in the dialogue."
+
+        err = self._validate_message_id(message)
+        if err:
+            return False, err
+
+        err = self._validate_message_target(message)
+        if err:
+            return False, err
+
+        return True, "The non-initial message passes basic validation."
+
+    def _validate_message_target(self, message: Message) -> Optional[str]:
+        """Check message target corresponds to messages in the dialogue, if not return error string."""
+        target = message.target
+        performative = message.performative
+
+        if (
+            message.message_id == self.STARTING_MESSAGE_ID
+            and target != self.STARTING_TARGET
+        ):
+            return "Invalid target. Expected 0. Found {}.".format(target)
```

Comment:
Because of this if block, this method now covers initial messages as well. Either specialise it to the non-initial case, or use this method in the `_basic_validation_initial_message()` method as well.

Reply:
updated, will be used in initial message validation